internal/pkg/deploy/cloudformation/lb_fargate_app_test.go

```diff
@@ -3,14 +3,145 @@
 
 package cloudformation
 
-import "testing"
+import (
+	"errors"
+	"os"
+	"testing"
+
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/archer"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/deploy"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/manifest"
+	"github.com/gobuffalo/packd"
+	"github.com/stretchr/testify/require"
+)
 
 func TestLBFargateStackConfig_StackName(t *testing.T) {
+	testCases := map[string]struct {
+		inAppName     string
+		inEnvName     string
+		inProjectName string
+
+		wantedStackName string
+	}{
+		"valid stack name": {
+			inAppName:     "frontend",
+			inEnvName:     "test",
+			inProjectName: "phonetool",
+
+			wantedStackName: "phonetool-test-frontend-app",
+		},
+		"longer than 128 characters": {
+			inAppName:     "whatisthishorriblylongapplicationnamethatcantfitintocloudformationwhatarewesupposedtodoaboutthisaaaaaaaaaaaaaaaaaaaa",
+			inEnvName:     "test",
+			inProjectName: "phonetool",
+
+			wantedStackName: "ol-test-whatisthishorriblylongapplicationnamethatcantfitintocloudformationwhatarewesupposedtodoaboutthisaaaaaaaaaaaaaaaaaaaa-app",
```

Comment:
Wait, really this is what we want? we cut off the front, not the back?

Reply:
:100: app name tho

Reply:
ahaha, the problem with cutting from the front is that we could end up cutting the application name completely and result with a string that's equal to the environment's stack. 

That's why I cut from the tail.